### PR TITLE
feat: disabled button

### PR DESCRIPTION
**Pull request by FredericoNoce**
### Related Issues

https://github.com/ZupIT/beagle/issues/1271

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/2a252e77f9ef6e5471341054dfb5e90f78167f29/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt#L43">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt#L43</a> - The function createButtonScreen is too long (64). The maximum length is 60.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by dantes-git — DISMISSED** [review]

**Review by gabrielleonardozup — APPROVED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt [hunk 1]
```diff
@@ -85,6 +89,14 @@ data class Button(
             }
         }
 
+        disabled?.let { bind ->
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt:92]: Shouldn't the button be able to see changes in this property even if it comes with the `disabled` not set?

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt [hunk 2]
```diff
@@ -58,49 +58,57 @@ object ButtonScreenBuilder {
...
+                Style(
+                    margin = EdgeValue(
+                        top = marginTop
+                    ))
+            ),
+            Button(
+                text = "Disabled Button",
+                styleId = "DesignSystem.Button.ScreenButton",
+                onPress = listOf(Alert(message = "This button must be disabled")),
+                disabled = true
```
Comment (matheusribeirozup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt:108]: What do you think of using this property from context to reactively enable and disable?
```suggestion
                disabled = expressionOf("@{context.disabled}")
```

**Review by carlossteinzup — DISMISSED** [review]

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt (hunk 2 above)
Comment (FredericoNoce) [reply backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ButtonScreenBuilder.kt:112]: done

**Review by carlossteinzup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/ButtonTest.kt [hunk 3]
```diff
@@ -144,4 +144,37 @@ class ButtonTest : BaseComponentTest() {
         // Then
         verify(exactly = 0) { analytics.trackEventOnClick(any()) }
     }
+
+    @Test
+    fun should_isEnabled_when_disable_null() {
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/components/ButtonTest.kt:149]: please follow style guide to write unit test: https://github.com/ZupIT/beagle/wiki/Android-Tests-Style-Guide

**Review by uziasferreirazup — CHANGES_REQUESTED** [review]

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/ButtonTest.kt (hunk 3 above)
Comment (FredericoNoce) [reply android/beagle/src/test/java/br/com/zup/beagle/android/components/ButtonTest.kt]: done!

**Review by uziasferreirazup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Pull request merged**